lib/gollum-lib/page.rb

```diff
@@ -17,29 +17,25 @@ class Page
     # Returns a Page
     attr_accessor :parent_page
 
-
-    # Checks a filename against the registered markup extensions
+    # Find the Symbol format for the file extension
     #
-    # filename - String filename, like "Home.md"
+    # ext - String file extension, like ".md"
     #
-    # Returns e.g. ["Home", :markdown], or [] if the extension is unregistered
-    def self.parse_filename(filename)
-      return [] unless filename =~ /^(.+)\.([a-zA-Z]\w*)$/i
-      pref, ext = Regexp.last_match[1], Regexp.last_match[2]
-
+    # Returns the Symbol format associated with ext
+    def self.parse_extension(ext)
       Gollum::Markup.formats.each_pair do |name, format|
```

Comment:
We want to know the `name` corresponding to the matching extension, not just whether there is a match. What we could do is rewrite `register` to generate a hash like this:
```ruby
formats = {
  'md': {name: :markdown},
  'mdown': {name: :markdown},
  'asciidoc': {name: :asciidoc},
  'creole': {name: :creole, reverse_links: true},
}
```
That way, `formats[ext]` would give us all we want to know.